code/socialDistribution/views.py

```diff
@@ -352,6 +354,28 @@ def posts(request, author_id):
     # In this case, app_name is socialDistribution
     return redirect('socialDistribution:home')
 
+# https://books.agiliq.com/projects/django-orm-cookbook/en/latest/copy.html - How to copy or clone an existing model object
+def sharePost(request, id):
+    """
+        Allows user to share a post.
+        The user that is sharing is the owner of the shared post
+        Public posts are shared to everyone
+        Friend posts are shared to friends
+    """
+    author = LocalAuthor.objects.get(user=request.user)
+    post = LocalPost.objects.get(id=id)
+    oldSource = post.get_id()
+    
+    post.pk = None # duplicate the post
+    post.author = author
+    post.published = timezone.now()
+    post.source = oldSource
+    post.save()
+    
+    dispatch_post(post, [])
```

Comment:
The shared post will maintain the same visibility right? How do we handle private posts? Based on this line, private posts won't be sent to anyone since the second param is []. Public posts should work though, right?

Reply:
Yep, added a new commit to only allow sharing of public/friend posts